modules/acm/src/test/scala/edu/gemini/epics/acm/ObserveStateSpec.scala

```diff
@@ -0,0 +1,402 @@
+  * care about the state of the channels. Instead we want to only observe
+  * the state transitions
+  */
+@SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements", "org.wartremover.warts.PublicInference"))
+final class ObserveStateSpec extends FunSuite with MockFactory {
+
+  test("NIFS normal observation") {
+    val context: CAJContext = mock[CAJContext]
+    (context.addContextExceptionListener _).expects(*).returns(()).repeat(5)
+    (context.addContextMessageListener _).expects(*).returns(()).repeat(5)
+    (context.pendIO _).expects(*).returns(()).repeat(1 to 6)
+    // We just return null as we don't need the channels and don't want to mock them
+    (context.createChannel(_: String)).expects("nifs:dc:nifsApply.DIR").returns(null)
+    (context.createChannel(_: String)).expects("nifs:dc:nifsApply.VAL").returns(null)
+    (context.createChannel(_: String)).expects("nifs:dc:nifsApply.MESS").returns(null)
+    (context.createChannel(_: String)).expects("nifs:dc:applyC.CLID").returns(null)
+    (context.createChannel(_: String)).expects("nifs:dc:applyC.VAL").returns(null)
+    (context.createChannel(_: String)).expects("nifs:dc:applyC.OMSS").returns(null)
+    val epicsService = new EpicsService(context)
+    val observe = new CaObserveSenderImpl(
+      "nifs::observeCmd",
+      "nifs:dc:nifsApply",
+      "nifs:dc:applyC",
+      "nifs:dc:observeC",
+      "nifs:dc:stop",
+      "nifs:dc:abort",
+      "NIFS Observe",
+      classOf[CarState],
+      epicsService)
+    // Start idle
+    assert(observe.applyState().isIdle())
+    // Post an observe
+    observe.post()
+
+    // CONFIGURATION
+    // APPLY Goes to busy
+    // VAL change
+    observe.onApplyValChange(358)
+    assert(!observe.applyState().isIdle())
+    // CAR CLID change
+    observe.onCarClidChange(358)
+    assert(!observe.applyState().isIdle())
+    // CAR VAL change
+    observe.onCarValChange(CarState.BUSY)
+    assert(!observe.applyState().isIdle())
+
+    // APPLY Goes to IDLE
+    // Another VAL change
+    observe.onApplyValChange(358)
+    assert(!observe.applyState().isIdle())
+    // CAR CLID change
+    observe.onCarClidChange(358)
+    assert(!observe.applyState().isIdle())
+    // CAR VAL change
+    observe.onCarValChange(CarState.IDLE)
+    assert(!observe.applyState().isIdle())
+
+    // OBSERVE
+    // OBSERVE goes BUSY
+    // VAL change
+    observe.onApplyValChange(359)
+    assert(!observe.applyState().isIdle())
+    // Observe CAR VAL change
+    observe.onObserveCarValChange(CarState.BUSY)
+    assert(!observe.applyState().isIdle())
+    // CAR CLID change
+    observe.onCarClidChange(359)
+    assert(!observe.applyState().isIdle())
+    // CAR VAL change
+    observe.onCarValChange(CarState.BUSY)
+    assert(!observe.applyState().isIdle())
+
+    // OBSERVE goes IDLE
+    // Observe CAR VAL change
+    observe.onObserveCarValChange(CarState.IDLE)
+    // And we are done and IDLE
```

Comment:
This is a bit odd, should we wait for apply to go IDLE before going back to idleState

Reply:
Yes, you are right